site/pages/concepts/multi-step-cast-actions.mdx

```diff
@@ -0,0 +1,233 @@
+
+```tsx twoslash [src/index.tsx]
+// @noErrors
+/** @jsxImportSource hono/jsx */
+import { Button, Frog, parseEther } from 'frog'
+import { abi } from './abi'
+
+export const app = new Frog()
+// ---cut---
+app.frame('/', (c) => {
+  return c.res({
+    image: (
+      <div style={{ color: 'white', display: 'flex', fontSize: 60 }}>
+        Add "Hello world!" Action
+      </div>
+    ),
+    intents: [
+      <Button.AddCastAction
+        action="/hello-world"
+      >
+        Add
+      </Button.AddCastAction>,
+    ]
+  })
+})
+
+app.castAction( // [!code focus]
+  '/hello-world', // [!code focus]
+  (c) => { // [!code focus]
+    console.log( // [!code focus]
+      `Cast Action to ${JSON.stringify(c.actionData.castId)} from ${ // [!code focus]
+        c.actionData.fid // [!code focus]
+      }`, // [!code focus]
+    ) // [!code focus]
+    return c.res({ type: 'frame', action: '/hello-world-frame' }) // [!code focus]
+  }, // [!code focus]
+  { name: "Hello world!", icon: "smiley" }) // [!code focus]
+) // [!code focus]
+```
+
+A breakdown of the `/hello-world` route handler:
+
+- `c.actionData` is never nullable and is always defined since Multi-step Cast Actions always do `POST` request.
+- We are responding with a `c.res` response and specifying a `message` that will appear in the success toast.
+
+### 3. Defining a frame handler
+Since in the previous step we send a response with `"type": "frame"` and have specified `"action": "/hello-world-frame"`, Client won't be able to resolve it since we have not defined it yet.
+
+So let's define one!
+
+```tsx twoslash [src/index.tsx]
+// @noErrors
+/** @jsxImportSource hono/jsx */
+// ---cut---
+import { Button, Frog, TextInput, parseEther } from 'frog'
+import { abi } from './abi'
+
+export const app = new Frog()
+
+app.frame('/', (c) => {
+  return c.res({
+    image: (
+      <div style={{ color: 'white', display: 'flex', fontSize: 60 }}>
+        Add "Hello world!" Action
+      </div>
+    ),
+    intents: [
+      <Button.AddCastAction action="/hello-world">
+        Add
+      </Button.AddCastAction>,
+    ]
+  })
+})
+
+app.castAction(
+  '/hello-world',
+  (c) => {
+    return c.res({ type: 'frame', action: '/hello-world-frame' })
+  },
+  { name: "Hello world!", icon: "smiley" })
+)
+
+app.frame('/hello-world-frame', (c) => { // [!code focus]
+  return c.res({ // [!code focus]
+    image: ( // [!code focus]
+      <div style={{ color: 'white', display: 'flex', fontSize: 60 }}> // [!code focus]
+        Hello world! // [!code focus]
+      </div> // [!code focus]
+    ) // [!code focus]
+  }) // [!code focus]
+}) // [!code focus]
+```
+
+Now we're all set! You can use existing Frame API to connect multiple frames together to have a multi-step experience. (see the [Connecting Frames (Actions)](/concepts/actions)).
+
+### 4. Bonus: Shorthand `c.frame`
+
+In order not to add property `"type": "frame"` to your `c.res(...)`, you can use a shorthand `c.frame(...)`.
```

Comment:
```suggestion
Instead of `c.res({ type: 'frame' })`, you can use a shorthand `c.frame(...)`.
```

Reply:
Need to apply the same in cast action doc